Add `--db-image` argument to `wasp deploy railway` (#3184)
wasp-lang/wasp: web/docs/deployment/deployment-methods/wasp-deploy/railway.md

- `my-wasp-app-client`
- `my-wasp-app-server`

> Railway doesn't allow setting the database service name using the Railway CLI. It will always be named `Postgres`. This also applies when using `--db-image`.

Reviewer: ```suggestion
Railway doesn't allow setting the database service name using the Railway CLI. It will always be named `Postgres`. This also applies when using the `--db-image` flag.
```

wasp deploy railway deploy <project-name>
```

#### Using a custom database image

> Use `--db-image <docker-image>` to provision the database with a specific Postgres image (for example, to enable extensions):

Reviewer: ```suggestion
Use the `--db-image <docker-image>` flag to set up the database with a specific Postgres image (for example, to enable extensions):
```

I think `set up` is easier understood than `provision`.

```shell
# PostGIS
wasp deploy railway launch my-wasp-app --db-image postgis/postgis

# pgvector
wasp deploy railway launch my-wasp-app --db-image pgvector/pgvector:pg16

> ```

Reviewer 2: I would turn these into two shell code blocks, since you don't want to run one and then the other.

> When provided, Wasp configures the DB service on Railway using the given image and required Postgres environment variables. If omitted, Railway’s default Postgres is used. The service name remains `Postgres`.

Reviewer: ```suggestion
If a custom database image is provided, Wasp configures the DB service on Railway using that image together with the required Postgres environment variables. Otherwise, Railway’s default Postgres image and environment variables are used. The service name always remains `Postgres`.
```

I'm not 100% sure with my suggestion.
I want to make it so that the sentence works without readying everything beforehand.

- `<project-name>-client`
- `<project-name>-server`

> Railway also creates a PostgreSQL database service named `Postgres` (the name remains the same even when using `--db-image`).

Reviewer: ```suggestion
Railway also creates a PostgreSQL database service named `Postgres`.
```

Do we need the brackets part here?
This is flag specific info which is explained in that flag's section.
I would assume the name is always the same even with above change.

> #### Using a custom database image

Reviewer 2: Sucks that we have all this repetition in docs. I sthere a way we could avoid it? Just link from one to another or sometihng? Maybe not?

> To create the database using a specific Postgres image, pass `--db-image <docker-image>`:

Reviewer: ```suggestion
To create the database using a specific Postgres image, pass the `--db-image <docker-image>` flag:
```

```shell
# PostGIS
wasp deploy railway setup my-wasp-app --db-image postgis/postgis

# pgvector
wasp deploy railway setup my-wasp-app --db-image pgvector/pgvector:pg16
```

> When provided, Wasp sets up the DB service with the specified image and configures necessary Postgres variables. If not provided, Railway’s default Postgres template is used.

Reviewer: Hm why is the explanation different here when this is exactly the same behavior as in `launch`.
`launch` just calls `setup`.
